src/HealthChecks.Uris/UriHealthCheck.cs

```diff
@@ -55,6 +59,25 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
                             return new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint #{idx} is not responding with code in {expectedStatusCodes.Min}...{expectedStatusCodes.Max} range, the current status is {response.StatusCode}.");
                         }
 
+                        if (expectedContent != null)
+                        {
+                            var actualContent = await response.Content.ReadAsStringAsync();
+                            if (actualContent != expectedContent)
+                            {
+                                return new HealthCheckResult(context.Registration.FailureStatus, description: $"Discover endpoint #{idx} is not responding with content {expectedContent}, the current content is {actualContent}.");
+                            }
+                        }
+
+                        if (expectedContentFunc != null)
+                        {
+                            var (isOk, notOkReason) = await expectedContentFunc(response.Content);

```

Comment:
If modify the code for use Func of string, you don't need this isOk

Reply:
I'm now returning strong typed results. Went down the path of just returning a string, but it seems too "loose". Types gives flexibility for future expansion too.